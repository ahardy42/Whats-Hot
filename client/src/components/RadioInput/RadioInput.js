import React from 'react';

const RadioInput = ({ handleCheck, isChecked, children, name, id, value, extraClasses}) => {
    return (
        <div className={`form-check form-check-inline ${extraClasses ? extraClasses : ""}`}>
            <input className={`form-check-input`} type="radio" name={name} id={id} value={value} checked={isChecked ? true : false} onClick={handleCheck}/>
            <label className={`form-check-label`} htmlFor={id}>{children}</label>
        </div>
    );
}

export default RadioInput;