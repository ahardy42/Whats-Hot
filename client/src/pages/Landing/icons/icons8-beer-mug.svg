<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 50 50" version="1.1" width="50" height="50" fill="#000000">
  <g id="surface1">
    <path style=" " d="M 10.3125 2 C 9.925781 2.074219 9.621094 2.367188 9.53125 2.75 C 9.53125 2.75 9 4.945313 9 9 L 9 45 C 9 46.644531 10.355469 48 12 48 L 34 48 C 35.644531 48 37 46.644531 37 45 L 37 40.5 C 37 40.203125 37.203125 40 37.5 40 C 41.230469 40 44.414063 38.355469 46.59375 35.71875 C 48.773438 33.082031 50 29.457031 50 25.5 L 50 21.90625 C 50 17.007813 45.992188 13 41.09375 13 L 40.15625 13 C 38.410156 13 37 11.589844 37 9.84375 L 37 9 C 37 4.910156 36.46875 2.75 36.46875 2.75 C 36.355469 2.308594 35.957031 2 35.5 2 L 10.5 2 C 10.46875 2 10.4375 2 10.40625 2 C 10.375 2 10.34375 2 10.3125 2 Z M 11.40625 4 L 34.59375 4 C 34.6875 4.421875 35 5.589844 35 9 L 11 9 C 11 5.621094 11.3125 4.425781 11.40625 4 Z M 11 11 L 35 11 L 35 43 L 11 43 Z M 23 12 C 20.800781 12 19 13.800781 19 16 C 19 18.199219 20.800781 20 23 20 C 25.199219 20 27 18.199219 27 16 C 27 13.800781 25.199219 12 23 12 Z M 12 12.15625 L 12 14.28125 C 12.59375 14.628906 13 15.261719 13 16 C 13 16.738281 12.59375 17.371094 12 17.71875 L 12 19.84375 C 13.722656 19.398438 15 17.859375 15 16 C 15 14.140625 13.722656 12.601563 12 12.15625 Z M 34 12.15625 C 32.277344 12.601563 31 14.140625 31 16 C 31 17.859375 32.277344 19.398438 34 19.84375 L 34 17.71875 C 33.40625 17.371094 33 16.738281 33 16 C 33 15.261719 33.40625 14.628906 34 14.28125 Z M 37 13.875 C 37.878906 14.566406 38.960938 15 40.15625 15 L 41.09375 15 C 44.910156 15 48 18.089844 48 21.90625 L 48 25.5 C 48 29.050781 46.90625 32.210938 45.0625 34.4375 C 43.21875 36.664063 40.648438 38 37.5 38 C 37.328125 38 37.164063 38.027344 37 38.0625 L 37 35.8125 C 37.304688 35.917969 37.628906 36 37.96875 36 C 40.535156 36 42.832031 34.660156 44.4375 32.5 C 46.042969 30.339844 47 27.363281 47 23.96875 L 47 22.40625 C 47 18.886719 44.113281 16 40.59375 16 L 39.40625 16 C 38.519531 16 37.691406 16.265625 37 16.71875 Z M 23 14 C 24.117188 14 25 14.882813 25 16 C 25 17.117188 24.117188 18 23 18 C 21.882813 18 21 17.117188 21 16 C 21 14.882813 21.882813 14 23 14 Z M 39.40625 18 L 40.59375 18 C 43.035156 18 45 19.964844 45 22.40625 L 45 23.96875 C 45 26.992188 44.148438 29.527344 42.84375 31.28125 C 41.539063 33.035156 39.839844 34 37.96875 34 C 37.421875 34 37 33.578125 37 33.03125 L 37 20.40625 C 37 19.058594 38.058594 18 39.40625 18 Z M 17 21 C 14.800781 21 13 22.800781 13 25 C 13 27.199219 14.800781 29 17 29 C 19.199219 29 21 27.199219 21 25 C 21 22.800781 19.199219 21 17 21 Z M 29 21 C 26.800781 21 25 22.800781 25 25 C 25 27.199219 26.800781 29 29 29 C 31.199219 29 33 27.199219 33 25 C 33 22.800781 31.199219 21 29 21 Z M 17 23 C 18.117188 23 19 23.882813 19 25 C 19 26.117188 18.117188 27 17 27 C 15.882813 27 15 26.117188 15 25 C 15 23.882813 15.882813 23 17 23 Z M 29 23 C 30.117188 23 31 23.882813 31 25 C 31 26.117188 30.117188 27 29 27 C 27.882813 27 27 26.117188 27 25 C 27 23.882813 27.882813 23 29 23 Z M 23 30 C 20.800781 30 19 31.800781 19 34 C 19 36.199219 20.800781 38 23 38 C 25.199219 38 27 36.199219 27 34 C 27 31.800781 25.199219 30 23 30 Z M 12 30.15625 L 12 32.28125 C 12.59375 32.628906 13 33.261719 13 34 C 13 34.738281 12.59375 35.371094 12 35.71875 L 12 37.84375 C 13.722656 37.398438 15 35.859375 15 34 C 15 32.140625 13.722656 30.601563 12 30.15625 Z M 34 30.15625 C 32.277344 30.601563 31 32.140625 31 34 C 31 35.859375 32.277344 37.398438 34 37.84375 L 34 35.71875 C 33.40625 35.371094 33 34.738281 33 34 C 33 33.261719 33.40625 32.628906 34 32.28125 Z M 23 32 C 24.117188 32 25 32.882813 25 34 C 25 35.117188 24.117188 36 23 36 C 21.882813 36 21 35.117188 21 34 C 21 32.882813 21.882813 32 23 32 Z M 17 39 C 15.140625 39 13.601563 40.277344 13.15625 42 L 15.28125 42 C 15.628906 41.40625 16.261719 41 17 41 C 17.738281 41 18.371094 41.40625 18.71875 42 L 20.84375 42 C 20.398438 40.277344 18.859375 39 17 39 Z M 29 39 C 27.140625 39 25.601563 40.277344 25.15625 42 L 27.28125 42 C 27.628906 41.40625 28.261719 41 29 41 C 29.738281 41 30.371094 41.40625 30.71875 42 L 32.84375 42 C 32.398438 40.277344 30.859375 39 29 39 Z M 11 45 L 35 45 C 35 45.566406 34.566406 46 34 46 L 12 46 C 11.433594 46 11 45.566406 11 45 Z "/>
  </g>
</svg>
