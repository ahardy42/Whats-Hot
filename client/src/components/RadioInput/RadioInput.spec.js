import React from 'react';
import renderer from 'react-test-renderer';
import RadioInput from './RadioInput';
import {ReactComponent as BeerMug} from '../../pages/Landing/icons/icons8-beer-mug.svg'
import { render, cleanup, fireEvent, waitForElement } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('the snapshot test', () => {
    it('renders a radio input', () => {
        const tree = renderer.create(
            <RadioInput
                id="test-radio"
                value="test"
                name="test-radio"
                handleCheck={jest.fn()}
                isChecked={true}
            >
                <BeerMug />
            </RadioInput>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})

describe('radio button unit test', () => {
    afterEach(() => {
        cleanup()
    });
    let isChecked = false;
    const mockClick = jest.fn(() => isChecked = !isChecked);
    it('renders a radio button', () => {
        const {container} = render(<RadioInput id="test" value="test" name="test" isChecked={isChecked} handleCheck={mockClick}><BeerMug/></RadioInput>)
        const input = container.querySelector('input');
        expect(input).toHaveAttribute('type', 'radio');
    });
    it('renders a label based on children', () => {
        const child = "some text";
        const {container} = render(<RadioInput id="test" value="test" name="test" isChecked={isChecked} handleCheck={mockClick}>{child}</RadioInput>);
        const label = container.querySelector('label');
        expect(label.textContent).toEqual(child)
    });
})