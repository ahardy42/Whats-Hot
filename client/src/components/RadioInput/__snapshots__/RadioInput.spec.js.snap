// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the snapshot test renders a radio input 1`] = `
<div
  className="form-check form-check-inline "
>
  <input
    checked={true}
    className="form-check-input"
    data-testid="test-radio"
    id="test-radio"
    name="test-radio"
    onChange={[MockFunction]}
    type="radio"
    value="test"
  />
  <label
    className="form-check-label"
    htmlFor="test-radio"
  >
    <svg>
      icons8-beer-mug.svg
    </svg>
  </label>
</div>
`;
